pulsar-broker/src/main/java/org/apache/pulsar/broker/service/SystemTopicBasedTopicPoliciesService.java

```diff
@@ -390,7 +390,6 @@ public void clean(TopicName topicName) {
             //change persistent://tenant/namespace/xxx-partition-0  to persistent://tenant/namespace/xxx
             realTopicName = TopicName.get(topicName.getPartitionedTopicName());
         }
-        policiesCache.remove(realTopicName);
```

Comment:
When the OwnerShip of Topic is transferred, we can not delete the cache, but when Topic is deleted, should this cache be deleted?

Reply:
Currently, the issue is when the topic been unloaded, we will not able to get the cache again if the namespace bundle ownership does not change. 

For the topic deletion case, I think we can write a message with key(topic name) and value of null to the system topic,  this will be compacted and we can also handle the null value message as the delete topic policy operation at the TopicPolicyService, I can push a separate PR to fix it.